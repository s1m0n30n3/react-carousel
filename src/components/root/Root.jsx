import React, { forwardRef } from "react";

export const Root = forwardRef(({ ...props }, ref) => (
  <div {...props} role="region" ref={ref} className="root--base" />
));

Root.displayName = "Gallery.Root";
