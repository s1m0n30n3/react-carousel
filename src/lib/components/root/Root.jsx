import React, { forwardRef } from "react";

import { base } from "styles/root.module.css";

export const Root = forwardRef(({ ...props }, ref) => (
  <div {...props} role="region" ref={ref} className={base} />
));

Root.displayName = "Gallery.Root";
