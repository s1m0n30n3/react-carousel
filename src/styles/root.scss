.root--base {
    width: 100%;
    height: 500px;
    background-color: #efefef;
    transform: translate3d(0,0,0);
}