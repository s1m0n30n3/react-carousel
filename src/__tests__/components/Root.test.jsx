import React from "react";
import renderer from "react-test-renderer";
import { describe, expect, test } from "vitest";

import { Root } from "../../components/root";

describe("Root", () => {
  test("component renders correctly without props", () => {
    const component = renderer.create(<Root />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
