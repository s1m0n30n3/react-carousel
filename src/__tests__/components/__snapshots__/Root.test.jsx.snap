// Vitest Snapshot v1

exports[`Root > component renders correctly without props 1`] = `
<div
  role="region"
/>
`;
